[Main] Move auth package to the clients to make it auth client, Auth Client Init Changes

caliskanugur opened the pull request:
- Moves auth package to clients
- Adds auth as a client to the rancher.Client - uses management client mainly, and uses the same session instantiated for the clients
- Adds a new function to refresh provider access group in users
- Adds update method to OLDAP

igomez06 reviewed (commented):
Needs an update after last PR, but LGTM

joesims22 reviewed (approved):
lgtm

anupama2501 commented:
Have a few conflicts. could you rebase please.

dasarinaidu reviewed (approved):
LGTM, please rebase and merge. Thanks

caliskanugur commented:
Rebased from 83 commits:

- Removed RBAC extensions accordingly since it is moved to actions.
- Updated the clients bits accordingly. 

Since there were so many changes that happened to both Shepherd and Clients, this PR is no longer safe to merge before the sanity checks.

After sanity checks for the tests that use this PR are provided, we can review this PR @dasarinaidu

igomez06 reviewed (approved):
LGTM

igomez06 reviewed (commented):
Will wait on sanity checks per Ugur's message
